src/Services/Types.tsx

```diff
@@ -0,0 +1,59 @@
+export type Todo = {
+  id: number,
+  title: string,
+  completed: boolean,
+};
+
+export enum Status {
+  All = 'All',
+  Active = 'Active',
+  Completed = 'Completed',
+}
+
+export type State = {
+  todos: Todo[],
+  filter: Status,
+};
+
+export type Action = ActionAdd
+| ActionDelete
+| ActionEdit
+| ActionComplete
+| AcctionToggleAll
+| ActionToggleFilter
+| ActionClearCompleted;
+
+type ActionAdd = {
+  type: 'add_todo',
```

Comment:
Define the available action types using an enum for better type safety. And think about a generic Action type

Reply:
Now i have very complex typification, is it worth?